fontforge/python.c

```diff
@@ -1734,51 +1734,105 @@ return( NULL );
 return( Py_BuildValue("i",ret));
 }
 
-static PyObject *PyFF_askChoices(PyObject *UNUSED(self), PyObject *args) {
-    char *title=NULL,*quest=NULL, **answers;
-    int def=0, cnt;
-    PyObject *answero;
-    int i, ret;
+static const char *askChoices_keywords[] = { "title", "question", "answers", "default", "multiple", NULL };
+
+static PyObject *PyFF_askChoices(PyObject *UNUSED(self), PyObject *args, PyObject *kwargs) {
+    char *title=NULL, *quest=NULL;
+    char **answers; // receives answers written into `answero`
+    PyObject* defo = NULL; // default answer index, or tuple of len cnt
+    int def = 0; // default index, only used if not multianswer
+    int cnt; // number of answers
+    PyObject *multipleo = NULL; // expected to be boolean
+    bool multiple = false;
+    PyObject *answero = NULL; // expected to be tuple
+    int ret;
 
     if ( no_windowing_ui ) {
-	PyErr_Format(PyExc_EnvironmentError, "No user interface");
-return( NULL );
+        PyErr_Format(PyExc_EnvironmentError, "No user interface");
+        return( NULL );
     }
 
-    if ( !PyArg_ParseTuple(args,"esesO|i","UTF-8", &title, "UTF-8", &quest, &answero, &def) )
-return( NULL );
+    if ( !PyArg_ParseTupleAndKeywords(args, kwargs, "esesO|OO", (char**) askChoices_keywords, 
+                                      "UTF-8", &title, "UTF-8", &quest, &answero, &defo, &multipleo) ) {
+        PyErr_Format(PyExc_EnvironmentError, "Failed to parse arguments");
+        return( NULL );
+    }
+
+    multiple = multipleo != NULL && PyObject_IsTrue(multipleo);
+
     if ( !PySequence_Check(answero) || STRING_CHECK(answero)) {
-	PyErr_Format(PyExc_TypeError, "Expected a tuple of strings for the third argument");
-	PyMem_Free(title);
-	PyMem_Free(quest);
-return( NULL );
+        PyErr_Format(PyExc_TypeError, "Expected a tuple of strings for the third argument");
+        PyMem_Free(title);
+        PyMem_Free(quest);
+        return( NULL );
     }
+
     cnt = PySequence_Size(answero);
-    answers = calloc(cnt+1, sizeof(char *));
-    if ( def<0 || def>=cnt ) {
-	PyErr_Format(PyExc_ValueError, "Value out of bounds for 4th argument");
-	PyMem_Free(title);
-	PyMem_Free(quest);
-return( NULL );
+    char* sel = calloc(cnt, sizeof(char));
+
+    if (defo != NULL && defo != Py_None) {
+        if ( PyInt_Check(defo) || PyLong_Check(defo) ) {
+            def = (int)PyLong_AsLong(defo);
+            if ( def<0 || def>=cnt ) {
+                PyErr_Format(PyExc_ValueError, "Value out of bounds for 4th argument");
+                PyMem_Free(title); PyMem_Free(quest); PyMem_Free(sel);
+                return( NULL );
+            }
+            if (multiple) {
+                sel[def] = (char)1;
+            }
+        } else if ( PyTuple_Check(defo) ) {
```

Comment:
I think it's fine to also allow the bool-tuple method for specifying the value in the single choice case. But if you do that you should verify that there is only one true value if `multiple` is not true.

Reply:
https://github.com/fontforge/fontforge/pull/4143/commits/dd31daf0eac65bfdfa5df5210a46292a97aa0080 :)